tests/InstrEngineTests/NaglerInstrumentationMethod/NaglerInstrumentationMethod.cpp

```diff
@@ -1187,6 +1202,133 @@ HRESULT CInstrumentationMethod::PerformSingleReturnInstrumentation(IMethodInfo*
     return S_OK;
 }
 
+HRESULT CInstrumentationMethod::AddExceptionHandler(IMethodInfo* pMethodInfo, IInstructionGraph* pInstructionGraph)
+{
+    HRESULT hr;
+
+    IModuleInfo* pModuleInfo;
+    IfFailRet(pMethodInfo->GetModuleInfo(&pModuleInfo));
+
+    IType* pReturnType;
+    IfFailRet(pMethodInfo->GetReturnType(&pReturnType));
+
+    CorElementType returnCorElementType;
+    IfFailRet(pReturnType->GetCorElementType(&returnCorElementType));
+
+    CComPtr<ILocalVariableCollection> localVars;
+    IfFailRet(pMethodInfo->GetLocalVariables(&localVars));
+
+    DWORD returnIndex = -1;
+    if (returnCorElementType != ELEMENT_TYPE_VOID)
+    {
+        IfFailRet(localVars->AddLocal(pReturnType, &returnIndex));
+    }
+
+    CComPtr<IMetaDataImport> spIMetaDataImport;
+    IfFailRet(pModuleInfo->GetMetaDataImport(reinterpret_cast<IUnknown**>(&spIMetaDataImport)));
+
+    CComPtr<IMetaDataAssemblyImport> spIMetaDataAssemblyImport;
+    IfFailRet(pModuleInfo->GetMetaDataImport(reinterpret_cast<IUnknown**>(&spIMetaDataAssemblyImport)));
+
+    mdAssembly resolutionScope;
+    spIMetaDataAssemblyImport->GetAssemblyFromScope(&resolutionScope);
+
+    std::wstring strExceptionTypeName(_T("System.Exception"));
```

Comment:
>std::wstring strExceptionTypeName(_T("System.Exception")); [](start = 4, length = 58)

We don't currently execute these tests on Linux, but I believe that std::wstring will give the wrong character width for calling functions that take a WCHAR in the ICorProfiler interfaces in Linux and Mac. We have another string type: tstring to handle this.

Reply:
Actually, never mind that comment. It seems that we don't handle it in general in this file. We should probably test and do a general cleanup.

---
In reply to: [566286936](https://github.com/microsoft/CLRInstrumentationEngine/pull/353#discussion_r566286936) [](ancestors = 566286936)

Reply:
Yes, it's technique I found used elsewhere.